types/wlr_screencopy_v1.c

```diff
@@ -90,6 +201,20 @@ static void frame_handle_output_precommit(struct wl_listener *listener,
 	frame_destroy(frame);
 }
 
+static void frame_handle_output_commit(struct wl_listener *listener,
```

Comment:
So this handler is just required to handle `screencopy_damage_accumulate` being called both from the damage precommit handler and from the frame precommit handler. Actually, can't we just always accumulate from both without this check? This will just add damage twice, which is the same as adding damage once.

Reply:
The accumulated damage is cleared at the end of the frame precommit callback. If the damage precommit callback happens after that, it will add damage to the accumulator that has already been cleared and sent to the client. This takes care of such an unfortunate sequence of events.

Reply:
Ah, right, there's this case too, you're right.

It's pretty annoying to be honest. We're just "lucky" to have two events (precommit then commit) here, and this `is_handled` flag is pretty complicated. Maybe we can find another solution? For instance, in the damage precommit handler do nothing if a frame object for the same output and client/connection exists?

Reply:
Yeah, I agree that it's a little complicated.
What about adding a new callback that reports damage that's always called before precommit? It would be three lines in a different part of the system versus a few lines plus added complexity in this part.